gpjax/parameters/module.py

```diff
@@ -0,0 +1,259 @@
+            if key not in fields:
+                raise ValueError(f"'{key}' is not a field of {type(self).__name__}")
+
+        pytree = copy(self)
+        pytree.__dict__.update(kwargs)
+        return pytree
+
+    def replace_meta(self, **kwargs: Any) -> Self:
+        """
+        Replace the metadata of the fields.
+
+        Args:
+            **kwargs: keyword arguments to replace the metadata of the fields of the object.
+
+        Returns:
+            Module: with the metadata of the fields replaced.
+        """
+        fields = vars(self)
+        for key in kwargs:
+            if key not in fields:
+                raise ValueError(f"'{key}' is not a field of {type(self).__name__}")
+
+        pytree = copy(self)
+        pytree.__dict__.update(_pytree__meta={**pytree._pytree__meta, **kwargs})
+        return pytree
+
+    def update_meta(self, **kwargs: Any) -> Self:
+        """
+        Update the metadata of the fields. The metadata must already exist.
+
+        Args:
+            **kwargs: keyword arguments to replace the fields of the object.
+
+        Returns:
+            Module: with the fields replaced.
+        """
+        fields = vars(self)
+        for key in kwargs:
+            if key not in fields:
+                raise ValueError(f"'{key}' is not a field of {type(self).__name__}")
+
+        pytree = copy(self)
+        new = deepcopy(pytree._pytree__meta)
+        for key, value in kwargs.items():
+            if key in new:
+                new[key].update(value)
+            else:
+                new[key] = value
+        pytree.__dict__.update(_pytree__meta=new)
+        return pytree
+
+    def replace_trainable(self: Module, **kwargs: Dict[str, bool]) -> Self:
+        """Replace the trainability status of local nodes of the Module."""
+        return self.update_meta(**{k: {"trainable": v} for k, v in kwargs.items()})
+
+    def replace_bijector(self: Module, **kwargs: Dict[str, Bijector]) -> Self:
+        """Replace the bijectors of local nodes of the Module."""
+        return self.update_meta(**{k: {"bijector": v} for k, v in kwargs.items()})
+
+    def constrain(self) -> Self:
+        """Transform model parameters to the constrained space according to their defined bijectors.
+
+        Returns:
+            Module: tranformed to the constrained space.
+        """
+
+        def _apply_constrain(meta_leaf):
+            meta, leaf = meta_leaf
+            return meta.get("bijector", Identity).forward(leaf)
+
+        return meta_map(_apply_constrain, self)
+
+    def unconstrain(self) -> Self:
+        """Transform model parameters to the unconstrained space according to their defined bijectors.
+
+        Returns:
+            Module: tranformed to the unconstrained space.
+        """
+
+        def _apply_unconstrain(meta_leaf):
+            meta, leaf = meta_leaf
+            return meta.get("bijector", Identity).inverse(leaf)
+
+        return meta_map(_apply_unconstrain, self)
```

Comment:
I find this confusing. What's the "base representation"? This makes me think that the bijector is effectively applied twice, i.e. `bijector.forward(bijector.forward(m.unconstrain())) == m.constrain()`, where I would expect the difference between "constrained" and "unconstrained" to be a _single_ application of the bijector :thinking: